catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/ws/SecureWebSocketServlet.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.ws;
+
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import org.codice.ddf.security.handler.api.HandlerResult;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
```

Comment:
We should use the JSR 356 API here instead also.

Reply:
I am not sure if this will work with `WebSocketServlet`.